scripts/xword/archives.lua

```diff
@@ -0,0 +1,116 @@
+local endswith = require 'pl.stringx'.endswith
+local join = require 'xword.join'
+
+-- Add the given handler for this file type if we don't know how to open it yet
+local function _add_handler_if_needed(test, constructor)
+    if wx.wxFileSystem.HasHandlerForPath(test) then
+        return
+    end
+    local h = constructor()
+    wx.wxFileSystem.AddHandler(h)
+    -- wxLua should ungc the handler in this function, but it doesn't.
+    -- If lua garbage collects this handler before we are done using it,
+    -- we'll crash.
+    wxlua.ungcobject(h)
+end
+
+-- Create a wxFileSystem with ReadFile, CopyFile, and CopyDirectory functions
+function xword.NewFs(archive)
```

Comment:
I know I wrote this (a long time ago), but especially now that it's top-level, this interface feels pretty rough. It's just a bunch of utility functions (add "#zip:" based on extensions, ReadFile, CopyFile, CopyDirectory) that pkgmgr needed.

This works, and I don't _really_ care that much, though all you should actually need to read a single zipped file is (a) something to register the wxFileSystemHandlers (just needs to happen once since it's global), and (b) that ReadFile function. In particular, the dependency on the xword pkgmgr join shim feels out of place.

:man_shrugging:

Reply:
Will work on simplifying the shared APIs, though it might mean a bit more code duplication with pkgmgr code. Let me see if I can get something clean with a simpler interface that pkgmgr can build on.

Reply:
This works for me. I was mostly lamenting my own old decisions, but IMO this code is certainly an improvement and `xword.archives` just has the archive-specific behavior, which makes sense to me.